x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/dataframe/extractor/DataFrameDataExtractorFactory.java

```diff
@@ -57,54 +58,71 @@
     private final String analyticsId;
     private final String index;
     private final ExtractedFields extractedFields;
+    private final QueryBuilder query;
```

Comment:
As mentioned above, if we apply the query whilst reindexing it doesn't seem necessary to use the query here at all.

Reply:
@dimitris-athanasiou it is enough to query on reindex. I think that is fine for now and I have made that change.

One concern with only using the query on the extractor is that only certain docs will get results in the index (the ones that match the query), while others simply don't have the results. This may cause some weirdness in user workflow, UI displaying, etc.

Reply:
I think in that case, a user that used a query should also be using that same query (including the UI) when viewing results unless they are intentionally doing something else.